File: InterviewBit/string/CountAndSay.py
class Solution:
    # @param A : integer
    # @return a strings=

    def countAndSayHelper(self, A):
        strA = str(A)
        dig0 = strA[0]
        i = 0
        currDig = dig0
        currCnt = 0
        res = ""
        while i<len(strA):
            #print currDig
            j=i
            while strA[j]==currDig:
                currCnt+=1
                j+=1
                if j==len(strA):
                    break


            res += str(currCnt)+currDig

            if j!=i and j<len(strA):
                currDig = strA[j]
                currCnt = 0
                i=j
            else:
                i=j+1
                currCnt=0


        return res
    def countAndSay(self,A):
        n = 1
        num = 1
        while (n<A):
            str_i = self.countAndSayHelper(num)
            num = int(str_i)
            print num
            n+=1
        return str(num)




if __name__=="__main__":
    sol = Solution()
    res = sol.countAndSay(4)
    print res

File: LeetCode/countAndSay.java
import java.util.*;
public class countAndSay {

    public String countAndSayHelper(String n) {
        String num = n;
        int i = 0;int j=0;
        String res="";
        while(i<num.length())
        {
            j=i;
            while(j<num.length() && num.charAt(j)==num.charAt(i))
            {
                j++;
            }
            res += String.valueOf(j-i)+(num.charAt(i)-'0');
            //System.out.println(res);
            i = j;
        }
        return res;
    }

    public String countAndSay(int n)
    {
        int i=1;
        if(n==i)
            return "1";

        int intRes = 1;
        String strRes="1";
        while(i<n)
        {
            strRes = countAndSayHelper(strRes);

            i++;
        }
        return strRes;
    }

    public static void main(String[] args) {
        int n=9;
        System.out.println(countAndSay(n));

    }

}
